crypto/multisig/src/v2.rs

```diff
@@ -0,0 +1,1102 @@
+                org1_signer2.public_key(),
+                org1_signer3.public_key(),
+                org2_signer1.public_key(),
+                org2_signer2.public_key(),
+                org2_signer3.public_key(),
+            ],
+        );
+    }
+
+    #[test]
+    fn ed25519_mixed_single_and_multi() {
+        let mut rng = Hc128Rng::from_seed([1u8; 32]);
+        let message = b"this is a test";
+
+        // Org 1 requires 2-of-3 signatures
+        let (org1_signerset, org1_signers) = make_signer_set(2, 3, &mut rng);
+        let (org1_signer1, org1_signer2, org1_signer3) =
+            (&org1_signers[0], &org1_signers[1], &org1_signers[2]);
+
+        // Org 2 requires 3-of-3 signatures
+        let (org2_signerset, org2_signers) = make_signer_set(3, 3, &mut rng);
+        let (org2_signer1, org2_signer2, org2_signer3) =
+            (&org2_signers[0], &org2_signers[1], &org2_signers[2]);
+
+        // Two single signers
+        let single_signer1 = Ed25519Pair::from_random(&mut rng);
+        let single_signer2 = Ed25519Pair::from_random(&mut rng);
+
+        // Sign the message with all of our signers.
+        let org1_signer1_sig = org1_signer1.try_sign(message.as_ref()).unwrap();
+        let org1_signer2_sig = org1_signer2.try_sign(message.as_ref()).unwrap();
+        let org1_signer3_sig = org1_signer3.try_sign(message.as_ref()).unwrap();
+
+        let org2_signer1_sig = org2_signer1.try_sign(message.as_ref()).unwrap();
+        let org2_signer2_sig = org2_signer2.try_sign(message.as_ref()).unwrap();
+        let org2_signer3_sig = org2_signer3.try_sign(message.as_ref()).unwrap();
+
+        let single_signer1_sig = single_signer1.try_sign(message.as_ref()).unwrap();
+        let single_signer2_sig = single_signer2.try_sign(message.as_ref()).unwrap();
+
+        // The top-level multisig requires 3-of-4 signatures
+        let signer_set = SignerSetV2::new(
+            vec![
+                org1_signerset.into(),
+                org2_signerset.into(),
+                single_signer1.public_key().into(),
+                single_signer2.public_key().into(),
+            ],
+            3,
+        );
+
+        // Signing with less than the threshold doesn't verify.
+        let multi_sig = MultiSig::new(vec![
+            // Valid org1 signature
+            org1_signer1_sig,
+            org1_signer2_sig,
+            // Invalid org2 signature
+            org2_signer1_sig,
+            org2_signer2_sig,
+            // One of the single signers
+            single_signer1_sig,
+        ]);
+        assert!(signer_set.verify(message.as_ref(), &multi_sig).is_err());
+
+        let multi_sig = MultiSig::new(vec![
+            // Valid org1 signature
+            org1_signer1_sig,
+            org1_signer2_sig,
+            org1_signer3_sig,
+            // A valid single signer
+            single_signer1_sig,
+        ]);
+        assert!(signer_set.verify(message.as_ref(), &multi_sig).is_err());
+
```

Comment:
:wrench: don't think this test case provides anything over the previous one or others in this file
```suggestion
```